### ISSUE TITLE
[loki] /tmp read-only file system

### ISSUE THREAD

**Issue by nick-oconnor** [grafana/helm-charts#609]
Similar issue to #577 with `/tmp`. I'm getting bombarded with `level=warn ts=2021-08-09T00:03:54.191957114Z caller=marker.go:214 msg="failed to process marks" path=/data/loki/boltdb-shipper-compactor/retention/markers/1628441994354518113 err="open /tmp/marker-view-656536243: read-only file system"`

I've worked around it by adding:
```
extraVolumes:
  - name: temp
    emptyDir: {}
extraVolumeMounts:
  - name: temp
    mountPath: /tmp
```
A+ for allowing extra volume mounts!

**Comment by pr0PM** [comment]
I'm also facing the exact same issue, but is this supposed to happen, needing us to mount extra, in that case we can add it directly in the loki-chart, or am I missing something?

Here is my loki-conf:
```
  auth_enabled: false
  chunk_store_config:
    max_look_back_period: 72h
  compactor:
    compaction_interval: 10m
    retention_delete_delay: 2h
    retention_delete_worker_count: 150
    retention_enabled: true
    shared_store: filesystem
    working_directory: /data/retention
  ingester:
    chunk_block_size: 262144
    chunk_idle_period: 60m
    chunk_retain_period: 1m
    chunk_target_size: 2097152
    lifecycler:
      ring:
        kvstore:
          store: inmemory
        replication_factor: 1
    max_transfer_retries: 0
  limits_config:
    cardinality_limit: 300000
    enforce_metric_name: false
    max_streams_per_user: 0
    reject_old_samples: false
    reject_old_samples_max_age: 168h
    retention_period: 72h
  schema_config:
    configs:
    - from: "2020-10-24"
      index:
        period: 24h
        prefix: index_
      object_store: filesystem
      schema: v11
      store: boltdb-shipper
  server:
    http_listen_port: 3100
  storage_config:
    boltdb:
      directory: /data/loki/index
    boltdb_shipper:
      active_index_directory: /data/loki/boltdb-shipper-active
      cache_location: /data/loki/boltdb-shipper-cache
      cache_ttl: 24h
      shared_store: filesystem
    filesystem:
      directory: /data/loki/chunks
  table_manager:
    retention_deletes_enabled: true
    retention_period: 72h
```

**Comment by varac** [comment]
Please reopen, since #717 only fixes this for `charts/loki-distributed`, but not for `charts/loki`.

**Comment by varac** [comment]
Getting this error in my logs still with latest loki chart v2.8.1.

**Comment by nick-oconnor** [comment]
@BitProcessor @dannykopping Yep, @varac is correct, #717 only addresses `loki-distributed`. The same fix needs to added to `loki`. Can you please reopen the issue? Thanks.

**Comment by edward-bear** [comment]
@BitProcessor Please implement for `charts/loki`, thanks.

**Comment by jortkoopmans** [comment]
This also applies to the loki-stack chart, which depends on the loki chart (currently at version ^2.3.0).

**Comment by almson** [comment]
What is wrong with the Kubernetes ecosystem that stuff like this can't be fixed in a timely fashion?

**Comment by BitProcessor** [comment]
Nothing is wrong @almson.
Feel free to open a PR with a fix, I’m sure it will get merged asap.

**Comment by almson** [comment]
@BitProcessor Sure, let's see if 10 minutes of my time is really the magic touch needed to getting major bugs fixed in this repository.

**Comment by almson** [comment]
@BitProcessor Nope, I was right.

@Bukashk0zzz You're welcome 🙂

**Comment by stefanandres** [comment]
This is actually broken again with the newest release. /tmp/ is missing again in https://github.com/grafana/helm-charts/pull/717/files

```
loki-0 single-binary level=warn ts=2022-11-21T11:13:48.191662555Z caller=marker.go:214 msg="failed to process marks" path=/var/loki/compactor/retention/markers/1669019389055641644 err="open /tmp/marker-view-2214401582: read-only file system"
loki-0 single-binary level=warn ts=2022-11-21T11:13:48.191681976Z caller=marker.go:214 msg="failed to process marks" path=/var/loki/compactor/retention/markers/1669019989059641868 err="open /tmp/marker-view-2565702123: read-only file system"
loki-0 single-binary level=warn ts=2022-11-21T11:13:48.191699959Z caller=marker.go:214 msg="failed to process marks" path=/var/loki/compactor/retention/markers/1669020589056171478 err="open /tmp/marker-view-3849071739: read-only file system"
loki-0 single-binary level=warn ts=2022-11-21T11:13:48.191717135Z caller=marker.go:214 msg="failed to process marks" path=/var/loki/compactor/retention/markers/1669021189056132868 err="open /tmp/marker-view-182832451: read-only file system"
loki-0 single-binary level=warn ts=2022-11-21T11:13:48.191739715Z caller=marker.go:214 msg="failed to process marks" path=/var/loki/compactor/retention/markers/1669021789059415454 err="open /tmp/marker-view-2415308346: read-only file system"
```

**Comment by juliagomezi** [comment]
I have two environments. One is reproducing this issue and the other one is not. I don't understand what provokes this issue. Why is it suddenly trying to write to /tmp when it hasn't before?

**Comment by Tazmaniac** [comment]
Please reopen.
A fix similar to gafana#7695 should be applied to production/helm/templates/backend/statefulset-backend.yaml to fix read-only tmp in 3 target mode (read,write,backend).
